File: app/src/main/java/comapps/com/sportsreference2/WebView.java
package comapps.com.sportsreference2;

import android.app.Activity;
import android.content.res.Configuration;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.util.DisplayMetrics;
import android.util.Log;
import android.view.Gravity;
import android.view.KeyEvent;
import android.view.View;
import android.view.WindowManager;
import android.webkit.WebViewClient;
import android.widget.Button;

/**
 * Created by me on 8/25/2015.
 */
public class WebView extends AppCompatActivity {

    private static final String TAG = "WEBVIEW";

    private android.webkit.WebView webView;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.webview);
    //    RelativeLayout layout = (RelativeLayout) findViewById(R.id.webLayout);

        Button closeButton = (Button) findViewById(R.id.closeButton);



        DisplayMetrics dm = new DisplayMetrics();
        getWindowManager().getDefaultDisplay().getMetrics(dm);

        int width = dm.widthPixels;
        int height = dm.heightPixels;


        Activity activity = WebView.this;
        int orientationValue = activity.getResources().getConfiguration().orientation;
        WindowManager.LayoutParams layoutParams = getWindow().getAttributes();
        WindowManager.LayoutParams mWindowParams = new WindowManager.LayoutParams();
        switch(orientationValue) {
            case Configuration.ORIENTATION_PORTRAIT:
                getWindow().setLayout((int) (width * .95), (int) (height * .80));
                layoutParams.y = (int) (height * .15);
                getWindow().setGravity(Gravity.TOP);
                getWindow().setAttributes(layoutParams);
                mWindowParams.gravity = Gravity.TOP;

                break;
            case Configuration.ORIENTATION_LANDSCAPE:
                getWindow().setLayout(width, (int) (height * .80));
                layoutParams.y = (int) (height * .25);
                getWindow().setGravity(Gravity.TOP);
                getWindow().setAttributes(layoutParams);
                mWindowParams.gravity = Gravity.TOP;

                break;

        }





        Bundle extras = getIntent().getExtras();





        String url = extras.getString("link");

        Log.d(TAG, "url is " + url);

        webView = (android.webkit.WebView) findViewById(R.id.webview1);


        webView.setWebViewClient(new WebViewClient());




        webView.getSettings().setJavaScriptEnabled(false);
        webView.getSettings().setLoadWithOverviewMode(false);
        webView.getSettings().setUseWideViewPort(true);
        webView.getSettings().setSupportZoom(true);
        webView.getSettings().setBuiltInZoomControls(true);
        webView.getSettings().setDisplayZoomControls(true);


       webView.setScrollBarStyle(View.SCROLLBARS_OUTSIDE_OVERLAY);
       webView.setScrollbarFadingEnabled(false);

        if (savedInstanceState != null) {
            webView.saveState(savedInstanceState);

        } else {

            if ( getResources().getConfiguration().orientation == 2 ) {
                webView.getSettings().setUserAgentString("Mozilla/5.0 (X11; U; Linux i686; en-US; rv:1.9.0.4) Gecko/20100101 Firefox/4.0");
                webView.loadUrl(url);
            } else {
                webView.loadUrl(url);
            }



        }





        closeButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {

                    destroyWebView();

            }
        });




    }

    private void destroyWebView() {



        if(webView != null) {
            webView.clearHistory();
            webView.clearCache(true);
            webView.loadUrl("about:blank");
            webView.pauseTimers();
            webView = null;

            finish();
        }

    }

    @Override
    public boolean onKeyDown(int keyCode, KeyEvent event) {

        if (keyCode == KeyEvent.KEYCODE_BACK){
            if(webView.canGoBack()){
                webView.goBack();
                return true;
            }
        }
        return super.onKeyDown(keyCode, event);
    }

    @Override
    protected void onSaveInstanceState(Bundle outState) {
        super.onSaveInstanceState(outState);
        webView.saveState(outState);
    }

    @Override
    protected void onRestoreInstanceState(Bundle savedState) {
        super.onRestoreInstanceState(savedState);
        webView.restoreState(savedState);
    }





}

File: app/src/main/java/comapps/com/sportsreference2/WebView.kt
package comapps.com.sportsreference2

import android.content.res.Configuration
import android.graphics.Color
import android.graphics.RectF
import android.net.Uri
import android.os.Build
import android.os.Bundle
import android.support.customtabs.CustomTabsIntent
import android.support.v7.app.AppCompatActivity
import android.util.DisplayMetrics
import android.util.Log
import android.view.Gravity
import android.view.View
import android.view.WindowManager
import com.google.gson.Gson
import com.google.gson.reflect.TypeToken


/**
 * Created by me on 8/25/2015.
 */
class WebView : AppCompatActivity() {

    private val TAG = "WEBVIEW_KOTLIN"

    private var customTabsOpened = false
    private var view: View? = null


    private lateinit var itemsHistory: ArrayList<SportsItem>


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.webview)
        //    RelativeLayout layout = (RelativeLayout) findViewById(R.id.webLayout);

        //      prefs = this.getSharedPreferences(
        //              "comapps.com.thenewsportsreference.app", Context.MODE_PRIVATE)

        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
            window.navigationBarColor = Color.BLACK
        }

        view = findViewById<View>(R.id.webview1)


        val dm = DisplayMetrics()
        windowManager.defaultDisplay.getMetrics(dm)


        val width = dm.widthPixels
        val height = dm.heightPixels

        val rf = RectF(2f, 2f, 2f, 2f)

        val radii = floatArrayOf(60f, 60f, 60f, 60f, 60f, 60f, 60f, 60f)

        val rrs = android.graphics.drawable.shapes.RoundRectShape(null, rf, radii)

        val able = android.graphics.drawable.ShapeDrawable(rrs)


        val activity = this@WebView
        val orientationValue = activity.resources.configuration.orientation
        val layoutParams = window.attributes
        val mWindowParams = WindowManager.LayoutParams()
        when (orientationValue) {
            Configuration.ORIENTATION_PORTRAIT -> {
                window.setLayout((width * .95).toInt(), (height * .90).toInt())
                layoutParams.y = (height * .05).toInt()
                window.setGravity(Gravity.TOP)
                window.attributes = layoutParams
                mWindowParams.gravity = Gravity.TOP
            }
            Configuration.ORIENTATION_LANDSCAPE -> {
                window.setLayout(width, (height * .80).toInt())
                layoutParams.y = (height * .25).toInt()
                window.setGravity(Gravity.TOP)
                window.attributes = layoutParams
                mWindowParams.gravity = Gravity.TOP
            }
        }// getWindow().setAttributes(layoutParams);
        // mWindowParams.gravity = Gravity.TOP;

        println("$TAG pre GSON sIO --> " + intent.extras.getString("sportItemObject"))

        val programIntent: String = intent.extras.getString("sentFrom", "")

        var sportsItemObject: SportsItem?

        try {
            sportsItemObject = Gson().fromJson(intent.extras.getString("sportsItemObject"),
                           SportsItem::class.java)
        } catch (e: Exception) {
            sportsItemObject = null
        }

        println("$TAG programIntent --> $programIntent")
        println("$TAG post GSON sIO --> ${sportsItemObject.toString()}")





        var webAddress = when (programIntent) {

            "ACTIVITY" -> intent.extras.getString("linkSearchText")

            "FAVORITE" -> intent.extras.getString("link")

            "HISTORY" -> sportsItemObject?.link

            "ADAPTER" -> sportsItemObject?.link

            else -> "http://www.sports-reference.com"
        }

        println("$TAG webAddress --> $webAddress")


        if (programIntent == "ADAPTER") {
            sportsItemObject?.let { addToHistoryList(it) }
        }



        if ( !webAddress!!.contains("http") ) {

            if (sportsItemObject!!.sport.contains("college") || webAddress.contains("cbb")) {
                webAddress = "https://www.sports-reference.com" + webAddress
            } else if (sportsItemObject.sport == "baseball") {
                webAddress = "https://baseball-reference.com" + webAddress
            } else if (sportsItemObject.sport == "basketball") {
                webAddress = "https://basketball-reference.com" + webAddress
            } else if (sportsItemObject.sport == "hockey") {
                webAddress = "https://hockey-reference.com" + webAddress
            } else if (sportsItemObject.sport == "football") {
                webAddress = "https://pro-football-reference.com" + webAddress
            } else if (sportsItemObject.sport.contains("soccer") || sportsItemObject.link.contains
                    ("squads")) {
                webAddress = "https://fbref.com" + webAddress
            }
        }


        val builder = CustomTabsIntent.Builder()
        builder.setToolbarColor(Color.BLACK)
        builder.enableUrlBarHiding()


        builder.setStartAnimations(this, R.anim.pushinfromright, R.anim.pushouttoleft)
        builder.setStartAnimations(this, R.anim.pushinfromleft, R.anim.pushouttoright)


        val customTabsIntent = builder.build()

        webAddress = webAddress.replace("`", "%27")



        Log.d("WEBVIEWKT", "url to load --> " + webAddress)




        customTabsIntent.launchUrl(this, Uri.parse(webAddress))
        customTabsOpened = true


        /*     webView = findViewById<View>(R.id.webview1) as android.webkit.WebView


             webView!!.webViewClient = WebViewClient()
             webView!!.webChromeClient = WebChromeClient()
             webView!!.background = able





             if (prefs.addFree) {
                 webView!!.settings.javaScriptEnabled = false
             } else {
                 webView!!.settings.javaScriptEnabled = true
                 webView!!.settings.domStorageEnabled = true
             }



             webView!!.settings.loadWithOverviewMode = false
             webView!!.settings.useWideViewPort = true
             webView!!.settings.setSupportZoom(true)
             webView!!.settings.builtInZoomControls = true
             webView!!.isScrollbarFadingEnabled = false
             webView!!.settings.displayZoomControls = true
             webView!!.scrollBarStyle = View.SCROLLBARS_OUTSIDE_OVERLAY


             if (savedInstanceState != null) {
                 webView!!.saveState(savedInstanceState)

             } else {

                 try {
                     webView!!.loadUrl(sportsItemObject.link)
                     addToHistoryList(sportsItemObject)
                 } catch (e: Exception) {
                     webView!!.loadUrl(sportsItemLink)
                 }

             }
     */


    }

    private fun addToHistoryList(o: SportsItem) {


        val type = object : TypeToken<java.util.ArrayList<SportsItem>>() {}.type
        itemsHistory = ArrayList()

        try {
            itemsHistory = Gson().fromJson<java.util.ArrayList<SportsItem>>(prefs.sih, type)
        } catch (e: Exception) {
        }

        println("WEBVIEWKT itemsHistory Json string --> ${prefs.sih}")
        println("WEBVIEWKT item to add to itemsHistory ----> $o")
        println("WEBVIEWKT itemsHistory.size --> ${itemsHistory.size}")

        if (prefs.sih.contains(o.link)) {
            println("WEBVIEWKT o in itemsHistory")
        } else {

            if (itemsHistory.size > 15) {
                itemsHistory.removeAt(15)
                itemsHistory.add(1, o)

            } else if (itemsHistory.size > 0) {
                itemsHistory.add(1, o)
            } else {
                itemsHistory.add(0, o)
            }

        }


        val jsonHistoryList = Gson().toJson(itemsHistory)
        prefs.sih = jsonHistoryList



        println("WEBVIEWKT itemsHistory.size --> ${itemsHistory.size}")
        println("WEBVIEWKT jsonHistoryList ---> ${prefs.sih}")

        //   mainActivityKotlin.onResume()
    }


    /*  private fun destroyWebView() {


          if (webView != null) {
              webView!!.clearHistory()
              webView!!.clearCache(true)
              webView!!.loadUrl("about:blank")
              webView!!.pauseTimers()
              webView = null

              finish()
          }

      }

      override fun onKeyDown(keyCode: Int, event: KeyEvent): Boolean {

          if (keyCode == KeyEvent.KEYCODE_BACK) {
              if (webView!!.canGoBack()) {
                  webView!!.goBack()
                  return true
              } else {

                  destroyWebView()
                  val intent = Intent(this, SportsItemActivityKotlin::class.java)
                  val gson = Gson()
               //   intent.putExtra("sportsItemObject", gson.toJson(listSportsItems[position]))
               //   Log.d("GSON", gson.toJson(listSportsItems[position]))
                  //    intent.putExtra("whichactivity", "baseball")
                  ContextCompat.startActivity(this, intent, null)
                  finish()
              }
          }
          return super.onKeyDown(keyCode, event)
      }

      override fun onSaveInstanceState(outState: Bundle) {
          super.onSaveInstanceState(outState)
          webView!!.saveState(outState)
      }

      override fun onRestoreInstanceState(savedState: Bundle) {
          super.onRestoreInstanceState(savedState)
          webView!!.restoreState(savedState)
      }*/

    override fun onPause() {

        finish()
        super.onPause()
    }

    override fun onBackPressed() {


    }


    companion object {

        private const val TAG = "WEBVIEW"
    }


}


